packages/block-editor/src/components/block-list-appender/style.scss

```diff
@@ -24,3 +30,13 @@
 .block-list-appender > .block-editor-inserter {
 	display: block;
 }
+
+
+// Hide the nested appender unless parent or child is selected.
```

Comment:
This PR was created just prior to https://github.com/WordPress/gutenberg/pull/20729, and did not account for "lighter DOM" blocks. Serendipitously, such a block was just merged, meaning the rebase caught an issue.

The selector is now slightly more complex, but I've added additional context to explain it.